lib/basedriver/driver.js

```diff
@@ -160,15 +166,96 @@ class BaseDriver extends Protocol {
     if (typeof eventName !== 'string') {
       throw new Error(`Invalid eventName ${eventName}`);
     }
-    if (!this._eventHistory[eventName]) {
-      this._eventHistory[eventName] = [];
+    if (!this._meta._eventHistory[eventName]) {
+      this._meta._eventHistory[eventName] = [];
     }
     const ts = Date.now();
     const logTime = (new Date(ts)).toTimeString();
-    this._eventHistory[eventName].push(ts);
+    this._meta._eventHistory[eventName].push(ts);
     log.debug(`Event '${eventName}' logged at ${ts} (${logTime})`);
   }
 
+  get instrumentation () {
+    return _.cloneDeep(this._meta._instrumentation.data);
+  }
+
+  wrapInstrumentation (eventName, fn) {
+    const spannerFunction = (...args) => {
+      this.instrumentEventStart(eventName);
+      let result;
+      try {
+        result = fn.call(...args);
+        this.instrumentEventEnd(eventName);
+      } catch (error) {
+        this.instrumentEventEnd(eventName);
+        throw error;
+      }
+      return result;
```

Comment:
if we return in the try, we don't need the separate `let` instantiation.

Reply:
done